### DISCUSSION TITLE
Duda sobre Mock/JUnit

### DISCUSSION THREAD

**Discussion by nubevi** [dds-jv/foro#243] [Consultas de herramientas] [answered]
Buenas!

Estoy tratando de verificar que mi mock "lowLevelFS" le llegue efectivamente el mensaje de "asyncReadFile".
Ahora, no sé si me esta fallando el test por el ultimo "Any()" en donde debería ir un Consumer<Integer> o es por otra cosa :P

Desde ya,

Muchas gracias

![image](https://user-images.githubusercontent.com/26943847/119911332-845d3080-bf2f-11eb-8749-8ea457f77fd0.png)

**Comment by nanderson87** [comment, marked as answer]
Hola @nubevi :

Efectivamente la excepción por null pointer esta sucediendo en esa linea. Proba por cambiarlo por:
`(anyInt(), any(), anyInt(), anyInt(), any(Consumer.class));`

Igual creo que el error viene más por los any cuando son int que el del consumer.

**Reply by nubevi** [reply]
Me tiro esto ahora :(

![image](https://user-images.githubusercontent.com/26943847/119918465-e7a28f00-bf3e-11eb-8df0-fcc4b96c97cd.png)

**Reply by nubevi** [reply]
La firma del metodo es la siguiente:

 void asyncReadFile(int fd, byte[] bufferBytes, int bufferStart, int bufferEnd,
                     Consumer<Integer> callback);

**Reply by nanderson87** [reply]
Ok. No esta pasando el chequeo de tipo. 
Cambiando ese solo por `any()` debería resultar ya que como dije en el comentario anterior me parece que te estaba tirando el null pointer  por usar  `any()` en los parámetros del tipo int que son primitivas y no objetos.

En resumen, probá asi:
(anyInt(), any(), anyInt(), anyInt(), any())

Segunda alternativa:
(anyInt(), any(), anyInt(), anyInt(), ArgumentMatchers.<Consumer<Integer>>any())

La primera debería funcionar (por lo menos, no marcarte el error detipo y no tirarte el null pointer).
